R/estimateAlpha.R

```diff
@@ -0,0 +1,274 @@
+              if(!(is.numeric(rarefaction.depth) & rarefaction.depth > 0)) {
+                  stop("'rarefaction.depth' must be a non-zero positive double.",
+                       call. = FALSE)
+                  }
+              # if multiple indices to be estimated, name should a vector of same length
+              if(length(index)!=length(name)) {
+                  stop("'index' and 'name' should be vectors of the same length.",
+                       call. = FALSE)
+                  }
+              # Looping over the vector of indices to be estimated
+              for (i in seq_along(index)) {
+                  # Getting the corresponding alpha indices function by parsing the index
+                  FUN <- NULL
+                  if(any(grepl(index[i], .get_indices("diversity")))) {
+                      # making name having the alpha type suffix or leave it as is if 
+                      # user defined
+                      name[i] <- .parse_name(index[i], name[i], "diversity")
+                      # cleaning index from suffix to be used with the corresponding index
+                      # function
+                      index[i] <- gsub("_diversity", "", index[i])
+                      FUN <- .estimate_diversity
+                  } else if (any(grepl(index[i], .get_indices("dominance")))) {
+                      name[i] <-  .parse_name(index[i], name[i], "dominance")
+                      index[i] <- gsub("_dominance", "", index[i])
+                      FUN <- .estimate_dominance
+                  } else if (any(grepl(index[i], .get_indices("evenness")))) {
+                      name[i] <- .parse_name(index[i], name[i], "evenness")
+                      if (index[i]!="simpson_evenness") {
+                          index[i] <- gsub("_evenness", "", index[i])
+                          }
+                      FUN <- .estimate_evenness
+                  } else if (any(grepl(index[i], .get_indices("richness")))) {
+                      name[i] <- .parse_name(index[i], name[i], "richness")
+                      index[i] <- gsub("_richness", "", index[i])
+                      FUN <- .estimate_richness
+                  } else {
+                  stop("'index' is coresponding to none of the alpha diversity indices.
+                  'index' should be one of: coverage_diversity, fisher_diversity,
+                  faith_diversity, gini_simpson_diversity,
+                  inverse_simpson_diversity, log_modulo_skewness_diversity,
+                  shannon_diversity, absolute_dominance, dbp_dominance,
+                  core_abundance_dominance, gini_dominance,
+                  dmn_dominance, relative_dominance, simpson_lambda_dominance,
+                  camargo_evenness, pielou_evenness, simpson_evenness,
+                  evar_evenness, bulla_evenness, ace_richness, chao1_richness,
+                  hill_richness or observed_richness.",
+                       call. = FALSE)
+                  }
```

Comment:
Check already in input check that every index can be found. There you can also parse the name. --> fail fast